app/models/user.rb

```diff
@@ -1,6 +1,18 @@
 class User < ActiveRecord::Base
   has_secure_password
 
+  before_validation :strip_whitespace
   validates :first_name, :last_name, :email, presence: true
   validates :email, uniqueness: true
+  validates :password, length: { minimum: 8 }
+  validates_format_of :email,
```

Comment:
Prefer the new style validations `validates :column, format: value` over `validates_format_of`.